### PR TITLE
ADD: [jni] add Activity and refactor to use our specific descendant

### PR DESCRIPTION
**Pull request by koying**
Add JNI Activity and makes us a descendant of that, as it should be rather than hacking directly inside Context

/cc @Montellese

### CODE REVIEW

File: xbmc/android/jni/Activity.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  static bool moveTaskToBack(bool nonRoot);
+
+private:
+  CJNIActivity();
+};
+
+///////////////////////////////////
+
+class CJNIXbmcMainActivity : public CJNIActivity
```
Comment (Montellese) [inline xbmc/android/jni/Activity.h]: Maybe remove the "Xbmc" from the class name.

File: xbmc/android/jni/Activity.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+{
+public:
+  CJNIXbmcMainActivity(const ANativeActivity *nativeActivity);
+  ~CJNIXbmcMainActivity();
+
+  static CJNIXbmcMainActivity* GetAppInstance() { return m_appInstance; }
+
+  static void _onNewIntent(JNIEnv *env, jobject context, jobject intent);
```
Comment (Montellese) [inline xbmc/android/jni/Activity.h]: I know this is copy & pasted but maybe change it to adhere to our coding guidelines i.e. `OnNewIntent`.

File: xbmc/android/jni/Activity.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+struct ANativeActivity;
+
+class CJNIActivity : public CJNIContext
+{
+public:
+  CJNIActivity(const ANativeActivity *nativeActivity);
+  ~CJNIActivity();
+
+  static bool moveTaskToBack(bool nonRoot);
```
Comment (Montellese) [inline xbmc/android/jni/Activity.h]: `MoveTaskToBack`

File: xbmc/android/jni/Activity.h (hunk 1 above)
Comment (koying) [inline xbmc/android/jni/Activity.h]: This is where all our specifics should go, so makes sense to me to have a specific name, doesn't it?

**Comment by Montellese** [comment]
Looks like a cleaner separation of the context and the activity.

File: xbmc/android/jni/Activity.h (hunk 2 above)
Comment (koying) [inline xbmc/android/jni/Activity.h]: This is the jni interface, coming from the JAVA classes.
davilla used this convention, but I can change it for all native functions, of course

File: xbmc/android/jni/Activity.h (hunk 3 above)
Comment (koying) [inline xbmc/android/jni/Activity.h]: moveTaskToBack is the actual name in SDK. We use the SDK names in the JNI library.

Comment (Montellese) [inline xbmc/android/jni/Activity.h]: Fair enough.

File: xbmc/android/jni/Activity.h (hunk 1 above)
Comment (Montellese) [inline xbmc/android/jni/Activity.h]: Yes but we are not XBMC anymore ;-)

File: xbmc/android/jni/Activity.h (hunk 2 above)
Comment (Montellese) [inline xbmc/android/jni/Activity.h]: Nah, in that case it's fine.

File: xbmc/android/jni/Activity.h (hunk 1 above)
Comment (koying) [inline xbmc/android/jni/Activity.h]: Right :)

File: xbmc/android/jni/Activity.h (hunk 2 above)
Comment (davilla) [inline xbmc/android/jni/Activity.h]: _onNewIntent is special, it is actually a java callback setup in android_main.cpp. Two other ones are _onReceive and _onFrameAvailable for "org/xbmc/xbmc/XBMCBroadcastReceiver" and "org/xbmc/xbmc/XBMCOnFrameAvailableListener".

**Comment by davilla** [comment]
ok and about time :)

**Comment by koying** [comment]
jenkins build and merge

**Pull request merged**